test/src/e2e_vm_tests/test_programs/should_fail/predicate_while/src/main.sw

```diff
@@ -0,0 +1,8 @@
+predicate;
+
+fn main() -> bool {
+    let mut x = 0;
+    while true { x += 1; }
```

Comment:
I'm curious to know if this still throws an error if the the while loop is used in a library function and that function is used in a predicate. If so, it's worth adding a test for it.

Reply:
Current solution was not throwing the error when the predicate uses a library with loops.
We still got the `The JI opcode cannot jump backwards in a predicate.` but it is not very useful because it has no span available.

We could check for while loops in other modules too but `type_check` checks all declarations so even functions that are not used would thrown an error if they contain loops. Ideally we want to throw an error only if the predicate uses a function with a while loop.

The solution was to perform the while in predicate check on `compile_while_loop` as at this time only used functions are compiled.